types/tls_test.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTLSVersion_PartialEq(t *testing.T) {
+	require.Greater(t, TLSv1_3, TLSv1_2)
+	require.Greater(t, TLSv1_2, TLSv1_1)
+	require.Greater(t, TLSv1_1, TLSv1_0)
+
+	require.Less(t, TLSv1_2, TLSv1_3)
+	require.Less(t, TLSv1_1, TLSv1_2)
+	require.Less(t, TLSv1_0, TLSv1_1)
+}
+
+func TestTLSVersion_Invalid(t *testing.T) {
+	require.NotEqual(t, TLSVersionAuto, TLSVersionInvalid)
```

Comment:
Is `TLSVersionAuto` the desired zero-value? If so, it might be worth making that explicit in a test:

```Go
var zeroValue TLSVersion
require.Equal(t, TLSVersionAuto, zeroValue)
```

Reply:
Initially I had thought it would be a more appropriate zero-value, but after some experimentation over in https://github.com/hashicorp/consul/pull/11576#discussion_r761496909 discovered `TLSVersionInvalid` would be the more appropriate zero-value, will add a test to cover that.